src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -361,12 +363,52 @@ private int ExploreTests()
                 runSettings[FrameworkPackageSettings.DefaultTestNamePattern] = options.DefaultTestNamePattern;
 
             if (options.TestParameters != null)
-                runSettings[FrameworkPackageSettings.TestParameters] = options.TestParameters;
+            {
+                var dictionary = new Dictionary<string, string>();
+
+                // Exactly what the framework used to do.
+                // Now can be independently changed without breaking backwards compatibility with old frameworks.
+                foreach (var param in options.TestParameters.Split(';'))
+                {
+                    var eq = param.IndexOf("=");
+
+                    if (eq > 0 && eq < param.Length - 1)
+                    {
+                        var name = param.Substring(0, eq);
+                        var val = param.Substring(eq + 1);
+
+                        dictionary[name] = val;
+                    }
+                }
+
+                SetTestParameters(runSettings, dictionary);
+            }
 
             return runSettings;
         }
 
         /// <summary>
+        /// Sets test parameters, handling backwards compatibility.
+        /// </summary>
+        private static void SetTestParameters(IDictionary<string, object> runSettings, IDictionary<string, string> testParameters)
+        {
+            runSettings[FrameworkPackageSettings.TestParametersDictionary] = testParameters;
+
+            if (testParameters.Count != 0)
+            {
+                // Kept for backwards compatibility with old frameworks.
+                // Reserializes the way old frameworks understand, even if the parsing above is changed.
+                // This reserialization cannot be changed without breaking compatibility with old frameworks.
+
+                var oldFrameworkSerializedParameters = new StringBuilder();
+                foreach (var parameter in testParameters)
```

Comment:
@CharliePoole I'm willing to follow https://github.com/nunit/docs/wiki/Coding-Standards#use-of-the-var-keyword and use `foreach (KeyValuePair<string, string> parameter in testParameters)` here too. Would if be good for me to update this?